lib/vdsm/common/exception.py

```diff
@@ -505,6 +505,18 @@ class MigrationOperationError(VdsmException):
     message = 'Error occurred during migration'
 
 
+class InvalidMPathConfiguration(InvalidConfiguration):
+    code = 103
+    message = "Invalid configuration value"
+
+    def __init__(self, reason, sections):
+        super().__init__(reason)
+        self.sections = sections
+
+    def __str__(self):
+        return f"{self.msg}: {self.context['reason']}"
```

Comment:
Why do we need to override `__str__`? The only reason I can see is include the `sections` in the log, but you don't include it.

The easiest way to handle this is to inherit the super class `__str__` and make sure the sections are included in the context, which is already part of the supervlass `__str__`.

A possible solution can be:

    class InvalidMPathConfiguration(ContextException):
        ...
        @property
        def sections(self):
            return self.context["sections"]

Now we have good `__str__` without writing any code, and we have a public property to access the sections that can be used by the tool to format a warning.

Reply:
The `__str__` method is overriden because the parent `__str__` returns a raw dictionary as in `{"reason": "this is the reason", "another_key": "another_value"}`, which is fine for the text of an exception but a bit ugly for the output of a command line tool.

Since I couldn't use the context dictionary from the tool where the exception is catched, I did override the string method.
I can expose the `reason` (and `context`) keys through properties as you suggest, but it is mostly a cosmetic discussion.